crypto/sparse_array.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include "internal/sparse_array.h"
+
+#ifdef OPENSSL_SMALL_FOOTPRINT
+ #define SA_BLOCK_BITS           4
+#else
+# define SA_BLOCK_BITS           12
```

Comment:
Sidenote: We could add a comment for both these bit lengths mirroring the rationale discussed in the PR comments

Reply:
I've added such a comment and allowed for compile time override of the setting.